launcher/experiments/experiments.go

```diff
@@ -0,0 +1,78 @@
+// Package experiments contains functionalities to retrieve syned experiments
+package experiments
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+// Experiments represents a collection of experiments that were synced.
+// All failures during the reading of experiment data are logged and returned.
+// This struct intends to be as safe as possible. Errors default to experiments being off, or not
+// existing. Failures should not cascade from this struct.
+// When using this class, if an experiment is not found, it should be assumed it is off.
+type Experiments struct {
+	experiments map[string]any
+	logger      *log.Logger
+}
+
+// NewExperiments returns an experiments.
+func NewExperiments(logger *log.Logger) Experiments {
+	return Experiments{
+		experiments: map[string]any{},
+		logger:      logger,
+	}
+}
+
+// GetExperiment returns an experiment value if it exists in the map.
+func (e Experiments) GetExperiment(key string) (any, error) {
+	val, ok := e.experiments[key]
+	if !ok {
+		return nil, fmt.Errorf("experiment not found for key %s", key)
+	}
+
+	return val, nil
+}
+
+// ReadExperimentsFile takes a filepath, opens the file, and calls ReadJsonInput with the contents
+// of the file.
+// If the file cannot be opened, the experiments map is set to an empty map.
+func (e *Experiments) ReadExperimentsFile(fpath string) error {
```

Comment:
Why doesn't the `New` function take the reader and do the initialization? This function seems unnecessarily exported.

Reply:
I wanted this class to contain the logic for opening the experiment file to be able to easily suppress errors.
To be able to test this code, I needed to pass in a reader.

This was my solution to accomplish both: I can test the reader method and consumers don't have to convert the file input into a reader to use this class.

Reply:
You can still test in the same package (`experiments_test.go`) without exporting. `New` can take a filepath and then call `readExperimentsFile`.

Reply:
Gotcha, made the change.